iceoryx_posh/source/roudi/application/roudi_app.cpp

```diff
@@ -67,37 +65,28 @@ void RouDiApp::registerSigHandler() noexcept
     sigemptyset(&act.sa_mask);
     act.sa_handler = roudiSigHandler;
     act.sa_flags = 0;
-    if (-1 == sigaction(SIGINT, &act, NULL))
+    if (cxx::makeSmartC(sigaction, cxx::ReturnMode::PRE_DEFINED_SUCCESS_CODE, {0}, {}, SIGINT, &act, nullptr)
+            .hasErrors())
     {
-        LogError() << "Calling sigaction() failed";
-        std::terminate();
+        LogFatal() << "Calling sigaction() failed";
+        errorHandler(Error::kROUDI_APP__COULD_NOT_REGISTER_SIGNALS, nullptr, ErrorLevel::FATAL);
     }
 
-    if (-1 == sigaction(SIGTERM, &act, NULL))
+    if (cxx::makeSmartC(sigaction, cxx::ReturnMode::PRE_DEFINED_SUCCESS_CODE, {0}, {}, SIGTERM, &act, nullptr)
+            .hasErrors())
     {
-        LogError() << "Calling sigaction() failed";
-        std::terminate();
+        LogFatal() << "Calling sigaction() failed";
+        errorHandler(Error::kROUDI_APP__COULD_NOT_REGISTER_SIGNALS, nullptr, ErrorLevel::FATAL);
     }
 
-    if (-1 == sigaction(SIGHUP, &act, NULL))
+    if (cxx::makeSmartC(sigaction, cxx::ReturnMode::PRE_DEFINED_SUCCESS_CODE, {0}, {}, SIGHUP, &act, nullptr)
+            .hasErrors())
     {
-        LogError() << "Calling sigaction() failed";
-        std::terminate();
+        LogFatal() << "Calling sigaction() failed";
+        errorHandler(Error::kROUDI_APP__COULD_NOT_REGISTER_SIGNALS, nullptr, ErrorLevel::FATAL);
```

Comment:
Please add a return after the `errorHandler`.

Reply:
done